pkg/volume/util/operationexecutor/operation_generator_test.go

```diff
@@ -0,0 +1,205 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package operationexecutor
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_model/go"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	"k8s.io/client-go/informers"
+	fakeclient "k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/tools/record"
+	utiltesting "k8s.io/client-go/util/testing"
+	controllervolumetesting "k8s.io/kubernetes/pkg/controller/volume/attachdetach/testing"
+	"k8s.io/kubernetes/pkg/features"
+	"k8s.io/kubernetes/pkg/volume"
+	"k8s.io/kubernetes/pkg/volume/csi"
+	"k8s.io/kubernetes/pkg/volume/gcepd"
+	volumetest "k8s.io/kubernetes/pkg/volume/testing"
+	volumetesting "k8s.io/kubernetes/pkg/volume/testing"
+	volumetypes "k8s.io/kubernetes/pkg/volume/util/types"
+	"os"
+	"testing"
+	"time"
+
+	featuregatetesting "k8s.io/component-base/featuregate/testing"
+	"k8s.io/kubernetes/pkg/volume/util"
+)
+
+const gcePdVolumePluginName = "kubernetes.io/gce-pd"
+
+type testcase struct {
+	name                  string
+	isCsiMigrationEnabled bool
+	expectedPluginName    string
+}
+
+// this method just tests the volume plugin name that's used in CompleteFunc, the same plugin is also used inside the
+// generated func so there is no need to test the plugin name that's used inside generated function
+func TestOperationGenerator_GenerateUnmapVolumeFunc_PluginName(t *testing.T) {
+
+	testcases := []testcase{
+		{
+			name:                  "csi migration disabled",
+			isCsiMigrationEnabled: false,
+			expectedPluginName:    gcePdVolumePluginName,
```

Comment:
just to be consistent can we use plugin defined in the csitranslation library?

Reply:
fixed, thanks!